src/backend/cuda/memory.cpp

```diff
@@ -102,17 +101,18 @@ void deviceMemoryInfo(size_t *alloc_bytes, size_t *alloc_buffers,
 
 template<typename T>
 T *pinnedAlloc(const size_t &elements) {
-    MemoryEventPair me =
-        pinnedMemoryManager().alloc(elements * sizeof(T), false);
+    buffer_info pair =
+        buffer_info(memoryManager().alloc(elements * sizeof(T), false));
+    event ev            = event(pair.getEvent());
+    Event &e            = getEvent(ev.get());
     cudaStream_t stream = getActiveStream();
```

Comment:
Its unfortunate that we don't have a wrapper around the queue objects. I tried to do that with the Event::QueueType typedef but the getActiveStream and the resulting objects have different APIs. We can refactor and combine a lot of this code later. Seems pretty repetitive.